vcsclient/github.go

```diff
@@ -342,10 +342,12 @@ func (client *GitHubClient) GetPullRequestByID(ctx context.Context, owner, repos
 		Source: BranchInfo{
 			Name:       sourceBranch,
 			Repository: *pullRequest.Head.Repo.Name,
+			Owner:      *pullRequest.Head.Repo.Owner.Login,
```

Comment:
We should check for nil values for all of this function... You can use `DefaultIfNotNil` func, under `vcsutils/utils.go`
Let's apply this function for all over